examples/url-shortener.php

```diff
@@ -101,12 +110,30 @@
   might happen here is the access token itself is
   refreshed if the application has offline access.
  ************************************************/
-if ($client->getAccessToken() && isset($_GET['url'])) {
-  $url = new Google_Service_Urlshortener_Url();
-  $url->longUrl = $_GET['url'];
-  $short = $service->url->insert($url);
-  $_SESSION['access_token'] = $client->getAccessToken();
+if ($client->getAccessToken() && isset($_REQUEST['url'])) {
+  if (validateCsrfToken()) {
+    $url = new Google_Service_Urlshortener_Url();
+    $url->longUrl = $_REQUEST['url'];
+    $short = $service->url->insert($url);
+    $_SESSION['access_token'] = $client->getAccessToken();
+  } else {
+    $invalidCsrf = true;
+    generateCsrfToken();
+  }
+}
+
+function generateCsrfToken()
+{
+  $_SESSION['csrf_token'] = bin2hex(openssl_random_pseudo_bytes(32));
+}
+
+function validateCsrfToken()
```

Comment:
This is great! However, we have a place for helper functions like this in [`examples/templates/base.php`](https://github.com/google/google-api-php-client/blob/master/examples/templates/base.php). Please move this function there.